pkg/hub/addon/addon.go

```diff
@@ -216,7 +222,12 @@ func (a *clusterProxyAddOnAgent) csrConfigurations(cluster *clusterv1.ManagedClu
 func (a *clusterProxyAddOnAgent) sign(csr *certificatesv1.CertificateSigningRequest) []byte {
 	// We consider approved csr no need to do verify again
 	klog.Infof("CSRSign, agent: %s", config.ADDON_AGENT_NAME)
-	s, err := a.hubKubeClient.CoreV1().Secrets(config.HUB_NAMESPACE).Get(context.TODO(), config.SignerSecret, metav1.GetOptions{})
+	hubNamespace, err := utils.GetComponentNamespace()
+	if err != nil {
+		klog.Errorf("get component namespace failed, err:%v", err)
```

Comment:
you might just want to warn here and fall back to use the open-cluster-management ns here?

Reply:
I think you may put the current component namespace in the `clusterProxyAddOnAgent` struct, when you new the addon agent, you pass the current namespace to it

Reply:
Sure. Using the `GetCurrentNamespace` in pkg helper with a default value now.